sdl2promela/stop_condition/translator.py

```diff
@@ -1112,6 +1183,29 @@ def _generate_filter_out_alternative(
         expression = _generate(context, statements[index].expression)
         expression = _add_precheck_expressions(context, expression)
         expressions.append(expression)
+    # For all observers with ignore state generate list of expressions
+    for observer in context.observers_with_ignore:
+        ignorestates = context.processes[observer].ignorestates
+        expression = _create_observer_state_check(ignorestates, observer)
+        expressions.append(expression)
+
+    if context.observers_with_success:
+        # Generate one expression AND for all observers with success state
+        # Join generated expression using AND
+        # Append result to list of expression
+        all_observer_expressions: List[promela.Expression] = []
```

Comment:
This code is repeated, right? Can it be extracted?

Reply:
Hmm, there's similar code, which wraps the expression using unary NOT, so there are two versions used in two places. I do not see a need to extract and parametrize it now, maybe in future